### DISCUSSION TITLE
Avoid using the same road twice

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#260] [Q&A] [answered]
This request is repeated a lot because it would be great to have it in Cruiser, but I couldn't find a separate topic.

Is it currently possible? GraphHopper seems to allow this (at least in the paid version). BRouter works wonders, but can such avoidance happen on its own or does it require special treatments?

**Comment by devemux86** [comment]
I have not seen a similar option in BRouter profiles,
even in https://github.com/poutnikl/Brouter-profiles.

cc @afischerdev @poutnikl

**Reply by poutnikl** [reply, marked as answer]
Information about passing a particular way the second time is not available on the profile level. It would have to be processed by routing application, or internally in BRouter preprocessing.

**Reply by devemux86** [reply]
@poutnikl thanks for the answer!

**Comment by devemux86** [comment]
Although this could also be easily done with some shaping points. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> Although this could also be easily done with some shaping points.

That's how I do it 🙂 Unfortunately, sometimes BRoutera tries very hard to make us take the same road again (of course I understand why it happens, but I don't want to do that 😉).

**Comment by umpfmpf** [comment]
I have used it in Kurviger and it has been very beneficial. Especially when you're driving, you don't have to worry about whether you've driven down the road before.
it would also be helpful when navigating if you weren't always guided back, but on a different route.

**Comment by Ciutnik** [comment]
On a micro scale, i.e. within one road, the solution would be to avoid U-turns. But it's probably not that easy to implement either. Maybe someday 🙂
